devito/core/autotuning.py

```diff
@@ -88,7 +87,7 @@ def autotune(operator, args, level, mode):
     # Perform autotuning
     timings = {}
     for n, tree in enumerate(trees):
-        blockable = [i.dim for i in tree if isinstance(i.dim, BlockDimension)]
+        blockable = [i.dim for i in tree if not is_integer(i.step)]
```

Comment:
It's integer only when you have += 1 increments, ie the inner loops that iterate within a block. So here we rule them out. 

In `master` we have `IncrDimension` and `BlockDimension`. With this PR, I generalised everything as `IncrDimension` (yet another improvement). But I can't use `isinstance(i.dim, IncrDimension)` here (for various reasons -- bear with me), so the alternative is to check for `is_integer`

Reply:
That's a confusing name if only +=1 is an integer (+=2 i an integer). Still don't get why "not is_integer" is the blockable one instead of "is_integer", as only the linear incr are blockable

Reply:
I would maybe agree with @mloubout here. Maybe we can choose a different name?

Reply:
I don't think we need to choose a different name

is this resolved?

Reply:
no problem